### ISSUE TITLE
!FindInMap issues while in Fn::ForEach, cannot find resource or parameter with DefaultValue

### ISSUE THREAD

**Issue by jburnham** [aws-cloudformation/cfn-lint#3241]
### CloudFormation Lint Version

0.87.2

### What operating system are you using?

Mac

### Describe the bug

Similar to https://github.com/aws-cloudformation/cfn-lint/issues/3198 except that when using FindInMap with a DefaultValue (added as an [language enhancement](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap-enhancements.html)), it still complains that `E1012: Ref AccountId not found as a resource or parameter` even though I would expect it to fall back to the default value in this case.

### Expected behavior

It should use the value of `DefaultValue` when the resource or parameter are not found.

### Reproduction template

Reusing the template in the previously reported bug with some slight modifications (hardcoded the AWS::AccountID instead of !Ref)

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::LanguageExtensions'

Description: Configures IAM Identity Center. IAM Identity Center must be enabled before you create this stack.

Parameters:

  IamIdentityCenterInstanceArn:
    Type: String
    Description: 'The ARN for the IAM Identity Center Instance. Found in the settings of IAM Identity Center.'

  IdentityStoreId:
    Type: String
    Description: 'The id for the identity store. Found in the settings of IAM Identity Center.'


Mappings:

  AccountIdsMap:
    # Commercial
    "101010101011":
      DevSandAccount: "1111111111112"
      ProjectADevAccount: "22222222222223"
      # ProejctADemoAccount: "333333333333334"


Resources:

  SystemAdminPermissionSet:
    Type: AWS::SSO::PermissionSet
    Properties:
      InstanceArn: !Ref IamIdentityCenterInstanceArn
      Name: 'SystemAdmin'
      Description: 'System Admin Permissions'
      SessionDuration: 'PT8H'
      InlinePolicy:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Action: '*'
            Resource: '*'

  SystemAdminGroup:
    Type: AWS::IdentityStore::Group
    Properties:
      DisplayName: SystemAdmin
      Description: 'System Admin Group'
      IdentityStoreId: !Ref IdentityStoreId

  # Testing Account Assignments
  'Fn::ForEach::SystemAdminAssignments':
    - AccountId
    - - DevSandAccount
      - ProjectADevAccount
      - ProejctADemoAccount
    - 'SystemAdminAssignment${AccountId}':
        Type: 'AWS::SSO::Assignment'
        Properties:
          InstanceArn: !Ref IamIdentityCenterInstanceArn
          PermissionSetArn: !GetAtt SystemAdminPermissionSet.PermissionSetArn
          TargetId: !FindInMap
            - AccountIdsMap
            - "101010101011"
            - !Ref AccountId
            - DefaultValue: "333333333333334"
          TargetType: AWS_ACCOUNT
          PrincipalId: !GetAtt SystemAdminGroup.GroupId
          PrincipalType: GROUP
```

`# ProejctADemoAccount: "333333333333334"` staying commented out causes the lint error. Re-enabling that line "corrects" the lint error, however this (I don't think) is ideal as it should use the default value so you don't have to define all cases in the map.

**Comment by kddejong** [comment]
Will be included in the next release
